lib/serializer.ts

```diff
@@ -135,9 +139,31 @@ function getSanitizedText(node: ts.Node, options: SerializationOptions): string
     }
   }
 
-  const children = node.getChildren();
+  let children = node.getChildren();
   if (children.length) {
-    return node.getChildren().map(n => getSanitizedText(n, options)).join('');
+    // Sort declarations under a class or an interface
+    if (node.kind === ts.SyntaxKind.SyntaxList) {
+      switch (node.parent && node.parent.kind) {
+        case ts.SyntaxKind.ClassDeclaration:
+        case ts.SyntaxKind.InterfaceDeclaration: {
+          if (children.every(node => node.kind in memberDeclarationOrder)) {
```

Comment:
The problem is that `SyntaxList` is arbitrarily generated by the `Node#getChildren()` function, so we have to somehow make sure we are working with the right list (of member declarations). We couldn't check all the sub-classes of `Declaration` either, thanks to duck typing.

How about replacing the `every` condition with:

```typescript
const members = node.parent.members;
if (children.every(node => members.indexOf(node) >= 0)) { ... }
```
Should be more future proof.